api/src/main/java/org/apache/brooklyn/api/internal/AbstractBrooklynObjectSpec.java

```diff
@@ -160,5 +172,9 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return Objects.hashCode(getCatalogItemId(), getDisplayName(), getType(), getTags());
     }
+
+    /** strings inserted as flags, config keys inserted as config keys; 
+     * if you want to force one or the other, create a ConfigBag and convert to the appropriate map type */
+    public abstract SpecT configure(Map<?,?> val);
```

Comment:
Configuration handling between specs seems identical these days, why not promote all related implementations?

But at least promote abstract `getFlags()/getConfig` as otherwise the javadoc doesn't make much sense, wdyt?